### PR TITLE
Certificate Manager

### PR DESCRIPTION
**Pull request by gambol99**
- adding the draft for the certificate manager

### CODE REVIEW

File: how-to-docs/cert-manager.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: tls
+spec:
+  secretName: tls
+  issuerRef:
+    name: platform-tls
+    kind: ClusterIssuer
+  commonName: site.svc.project.cluster.local
```
Comment (jaykeshur) [inline how-to-docs/cert-manager.md]: site.namespace.svc.cluster.local

File: how-to-docs/cert-manager.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+kind: Certificate
+metadata:
+  name: tls
+spec:
+  secretName: tls
+  issuerRef:
+    name: project-ca
+    # @Note: we have change from ClusterIssuer to a local Issuer
+    kind: Issuer
+  commonName: site.svc.project.cluster.local
```
Comment (jaykeshur) [inline how-to-docs/cert-manager.md]: see previous comment

File: how-to-docs/cert-manager.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+Lets assume we have an externally facing site which we wish to expose via ingress and we want a valid Letsencrypt certificate. Two have two things to note here;
+
+- the enable annotation `certmanager.k8s.io/enabled` which is a toggle to ask cert-manager to handle this ingress resource.
+- the Letsencrypt challenge `certmanager.k8s.io/acme-challenge-type`; note, technically the challenge is not required as `http01` is the default type, but I'm adding to highlight.
+
+**When to use a HTTP challenge**
+
+Assuming the site is externally facing i.e. the ingress class on the ingress is `kubernetes.io/ingress.class: ingress-external` you should always default to using a http01 challenge. Things change, however, when the site is internal / behind the vpn. In order to handle the challenge when behind the VPN you need to switch to using a DNS challenge. In ingress this is done by adding the annotation `certmanager.k8s.io/acme-challenge-type: dns`. **Very Important** this assumes you have contacted the ACP beforehand and we have either added the domain to our route53 or added the dns provider as a challenge provider.
```
Comment (jaykeshur) [inline how-to-docs/cert-manager.md]: nginx-external

File: how-to-docs/cert-manager.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  issuerRef:
+    kind: ClusterIssuer
+    name: letsencrypt-prod
+  secretName: example-tls
+```
+
+You can review, get, list and describe the Certificate like any other kubernetes resource within your namespace.
+
+```shell
+k -n project get certificate
```
Comment (jaykeshur) [inline how-to-docs/cert-manager.md]: use kubectl

File: how-to-docs/cert-manager.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+```YAML
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: example-tls
+spec:
+  secretName: example-tls
+  issuerRef:
+    name: letsencrypt-prod
+  commonName: '*.example.com'
```
Comment (jaykeshur) [inline how-to-docs/cert-manager.md]: Are we recommending wildcards? Think we should stick to explicit names.

**Comment by gambol99** [comment]
updated @marcinc

**Pull request merged**